cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaReconciler.java

```diff
@@ -1020,6 +1029,56 @@ protected Future<Void> addListenersToKafkaStatus(KafkaStatus kafkaStatus) {
+        kafkaStatus.setKafkaMetadataState(kafkaMetadataStateManager.computeNextMetadataState(kafkaStatus).name());
+        return Future.succeededFuture();
+    }
+
+    /**
+     * This method checks if a migration is still ongoing on the Kafka side, through the KafkaMetadataStateManager instance.
+     * A ZooKeeper to KRaft migration can take some time and, on each reconcile, the operator checks its status by calling this method.
+     * Internally, the KafkaMetadataStateManager instance is leveraging the endpoint exposed by the Kafka Agent which provides
+     * the KRaft migration state through a corresponding metric.
+     *
+     * @return  Future which completes when the check on the migration is done
+     */
+    protected Future<Void> updateKafkaMetadataMigrationState() {
+        KafkaMetadataConfigurationState kafkaMetadataConfigState = this.kafkaMetadataStateManager.getMetadataConfigurationState();
+        // on each reconcile, would be useless to check migration status if it's not going on
+        if (kafkaMetadataConfigState.isMigration()) {
+            return ReconcilerUtils.clientSecrets(reconciliation, secretOperator)
+                    .compose(compositeFuture -> {
+                        // we should get the quorum controller leader using the Admin Client API describeMetadataQuorum() but
+                        // it's not supported by brokers during migration because they are still connected to ZooKeeper so ...
+                        // going through the controllers set to get metrics from one of them, because all expose the needed metrics
+                        boolean zkMigrationStateChecked = false;
+                        for (NodeRef controller : kafka.controllerNodes()) {
+                            try {
+                                LOGGER.infoCr(reconciliation, "Checking ZooKeeper migration state on controller {}", controller.podName());
+                                KafkaAgentClient kafkaAgentClient = kafkaAgentClientProvider.createKafkaAgentClient(
+                                        reconciliation,
+                                        compositeFuture.resultAt(0),
+                                        compositeFuture.resultAt(1)
+                                );
+                                this.kafkaMetadataStateManager.setMigrationDone(
+                                        KRaftMigrationUtils.checkMigrationInProgress(
+                                                reconciliation,
+                                                kafkaAgentClient,
+                                                controller.podName()
+                                        ));
+                                zkMigrationStateChecked = true;
+                                break;
+                            } catch (RuntimeException e) {
+                                LOGGER.warnCr(reconciliation, "Error on checking ZooKeeper migration state on controller {}", controller.podName());
```

Comment:
I don't know ... I can imagine it being debug. If it is a warning, maybe you should explain that other node might be tried?

Reply:
Better moving to DEBUG then. It will be consistent with the previous log where we are "Checking ZooKeeper migration state ...." Otherwise the user would see just a WARN but not the "cheking" message if DEBUG level is not enabled.